### DISCUSSION TITLE
A way to access or retrieve bulk game data

### DISCUSSION THREAD

**Discussion by InterestingBrainPoops** [BattlesnakeOfficial/feedback#170] [Feature Requests]
Something that I've come across a lot when thinking of ways to improve my snake is the lack of game data to test / train it on.  
I don't know how widespread this issue is, but some way to get the states / data for games that have been run on the battlesnake servers would be amazing.  
Ie some download link on the website or some other way to get this kind of data.  
It would also be really cool to run some analysis on what is considered a good / bad board state, and what parameters really matter.  
As of right now, people have to run their own games or log the game states to get data, and even then its really time consuming to actually generate a good dataset.  
Assuming you have 1 snake running in the non-beta global arena, you would have:
2.7 snakes in the game on average,  
300 turns per game on average,  
1 game per 10 minutes.  
That means to get a decent dataset (1mil states , (battlesnake has a lot of complexity)),  you would need around 9 days.

**Comment by mawilson** [comment]
I've been writing tests by squinting at games I watched & copying over coordinates manually to replicate the game state. Not sure if a smarter way already exists, but I would definitely appreciate a way to download a collection of game states from a game.

**Reply by InterestingBrainPoops** [reply]
For individual game states, you can use https://github.com/TheApX/battlesnake-engine-cpp/tree/main/gamedownloader or the https://addons.mozilla.org/en-US/firefox/addon/battlesnake-downloader/ addon by l4r0x. They both work, and I personally use the second because its extremely convinient.

**Reply by mawilson** [reply]
Neat, thanks. Didn't mean to hijack your idea, a bulk game downloader would be pretty rad.

**Comment by tphummel** [comment]
I've been exporting decision data on each request. As simple as writing out a log file depending on how you are hosted. I used it as an opportunity to try honeycomb.io but anything you can hook in once you send off the response could work. Of course this would only help going forward and not super helpful to capture the existing history. for example: https://github.com/tphummel/bobby-witt/blob/main/index.js#L299
